src/VisualStudio/Core/Def/Implementation/DebuggerIntelliSense/DebuggerIntellisenseFilter.cs

```diff
@@ -120,24 +106,20 @@ void executeNextCommandTarget()
                 // legacy stuff they interfaced with. That means we get SCROLLUP if the user
                 // types escape, so treat SCROLLUP like CANCEL. It's actually a CANCEL.
                 case VSConstants.VSStd2KCmdID.SCROLLUP:
-                    ExecuteCancel(subjectBuffer, contentType, () =>
-                    {
-                        // We cannot just pass executeNextCommandTarget becuase it would execute SCROLLUP
-                        var cancelCmdGroupId = VSConstants.VSStd2K;
-                        NextCommandTarget.Exec(ref cancelCmdGroupId, (uint)VSConstants.VSStd2KCmdID.CANCEL, executeInformation, pvaIn, pvaOut);
-                    });
+                    NextCommandTarget.Exec(VSConstants.VSStd2K, (uint)VSConstants.VSStd2KCmdID.CANCEL, executeInformation, pvaIn, pvaOut);
                     break;
 
                 // If we see a RETURN, and we're in the immediate window, we'll want to rebuild
                 // spans after all the other command handlers have run.
                 case VSConstants.VSStd2KCmdID.RETURN:
-                    ExecuteReturn(subjectBuffer, contentType, executeNextCommandTarget);
+                    result = NextCommandTarget.Exec(ref pguidCmdGroup, commandId, executeInformation, pvaIn, pvaOut);
```

Comment:
The existing code here has some subtle bugs: the Execute* methods that took `executeNextCommandTarget` *would* update `result`, but the code that didn't would fail to update result and we'd always return S_OK. At this point, I'm leaving the bugs in place.